src/idp/verified_request.rs

```diff
@@ -0,0 +1,48 @@
+use crate::schema::AuthnRequest;
+use crate::crypto::{self, verify_signed_xml};
+
+use super::error::Error;
+use std::str::FromStr;
+
+pub struct UnverifiedAuthnRequest<'a> {
+    pub request: AuthnRequest,
+    xml: &'a str,
+}
+
+impl <'a> UnverifiedAuthnRequest<'a> {
+    pub fn from_xml(xml: &str) -> Result<UnverifiedAuthnRequest, Error> {
+        Ok(UnverifiedAuthnRequest {
+            request: AuthnRequest::from_str(xml)?,
```

Comment:
```suggestion
            request: xml.parse()?,
```
This also implements `FromStr`